sql/core.go

```diff
@@ -407,6 +439,10 @@ type StatisticsTable interface {
 	NumRows(*Context) (uint64, error)
 	// DataLength returns the length of the data file (varies by engine).
 	DataLength(ctx *Context) (uint64, error)
+	// CalculateStatistics fills in the histogram object for the associate table.
+	CalculateStatistics(ctx *Context) error
```

Comment:
Call this AnalyzeTable

Comment: `AnalyzeTable is a hook to update any cached or persisted statistics for this table. It may be triggered by an analyze table statement, or automatically when the engine decides it is necessary. Integrators can ignore this hook and implement their own method of keeping statistics up to date, at the cost of potentially stale statistics.`